commit 3e8a0a559c66ee9e7468195691a56fefc3589740
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date:   Wed Aug 13 13:48:39 2008 -0700

    dccp: change L/R must have at least one byte in the dccpsf_val field
    
    Thanks to Eugene Teo for reporting this problem.
    
    Signed-off-by: Eugene Teo <eugenete@kernel.sg>
    Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
    Signed-off-by: Gerrit Renker <gerrit@erg.abdn.ac.uk>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Adjusted to apply to Debian's 2.6.24 by dann frazier <dannf@hp.com>

diff -urpN linux-source-2.6.24.orig/net/dccp/proto.c linux-source-2.6.24/net/dccp/proto.c
--- linux-source-2.6.24.orig/net/dccp/proto.c	2008-01-24 15:58:37.000000000 -0700
+++ linux-source-2.6.24/net/dccp/proto.c	2008-08-26 16:07:29.000000000 -0600
@@ -458,6 +458,11 @@ static int dccp_setsockopt_change(struct
 
 	if (copy_from_user(&opt, optval, sizeof(opt)))
 		return -EFAULT;
+	/*
+	 * rfc4340: 6.1. Change Options
+	 */
+	if (opt.dccpsf_len < 1)
+		return -EINVAL;
 
 	val = kmalloc(opt.dccpsf_len, GFP_KERNEL);
 	if (!val)
